crates/e2e/tests/e2e/protocol_fee.rs

```diff
@@ -281,10 +92,18 @@ async fn surplus_fee_buy_order_test(web3: Web3) {
         factor: 0.3,
         max_volume_factor: 0.9,
     };
-    let protocol_fee = ProtocolFee {
-        policy: fee_policy,
-        policy_order_class: FeePolicyOrderClass::Market,
-    };
+    let protocol_fees = vec![
+        ProtocolFee {
+            policy: fee_policy.clone(),
+            policy_order_class: FeePolicyOrderClass::Market,
+        },
+        // Tests ability of providing multiple fees in the config. In fact, the Market one is used
+        // only since the order is in-market.
+        ProtocolFee {
+            policy: fee_policy,
+            policy_order_class: FeePolicyOrderClass::Limit,
+        },
+    ];
```

Comment:
How about making a single test case that contains a limit order with surplus fee and a market order with price improvement fee (and potentially a third order with partner fee) in a batch to make sure all fee models that are used in practice work as expected?

Reply:
I tried to create an out-of-market order, but is there an easy way to fill a non-partial out-of-market order? `limit_orders.rs`, for example, contains only in-market orders. If not, this is a more involved change to include partially fillable orders in the suite, and I am already considering tackling it in the cope of #2468(probably after protocol fee release and 0x liquidity).

Reply:
> Also, not sure how to keep [this assertion](https://github.com/cowprotocol/services/blob/fc910f30b2212072e2c69e142e4ca9ed2e5ecf32/crates/e2e/tests/e2e/protocol_fee.rs#L339-L351) with a single batch of multiple orders

How about using different tokens for the different orders?

> I tried to configure out-of-market order and couldn't make it work yet.

The easiest way to make an out of market order "work" IMO is to create an AMM with balances A and B, get a quote and place an order this is "out of market" given that quote. Then you send a ton of A into the AMM (making the price change and the order now being executable).